src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_EditProfileScreen.tsx

```diff
@@ -0,0 +1,324 @@
+import {ProfileScreenProps} from '@atb/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/navigation-types';
+import {ActivityIndicator, View} from 'react-native';
+import {Section, TextInputSectionItem} from '@atb/components/sections';
+import React, {useCallback, useEffect, useState} from 'react';
+import {useTranslation} from '@atb/translations';
+import {EditProfileTexts} from '@atb/translations/screens/subscreens/EditProfileScreen';
+import {FullScreenView} from '@atb/components/screen-view';
+import {ThemeText} from '@atb/components/text';
+import {StyleSheet} from '@atb/theme';
+import {useAuthState} from '@atb/auth';
+import {Button} from '@atb/components/button';
+import Delete from '@atb/assets/svg/mono-icons/actions/Delete';
+import {MessageBox} from '@atb/components/message-box';
+import {emailAvailable, getProfile, updateProfile} from '@atb/api';
+import parsePhoneNumber from 'libphonenumber-js';
+import {ArrowRight} from '@atb/assets/svg/mono-icons/navigation';
+import {CustomerProfile} from '@atb/api/types/profile';
+import {numberToAccessibilityString} from '@atb/utils/accessibility';
+
+type EditProfileScreenProps = ProfileScreenProps<'Profile_EditProfileScreen'>;
+type SubmissionStatus =
+  | 'loading'
+  | 'invalid-email'
+  | 'unavailable-email'
+  | 'submitted'
+  | 'submission-error';
+type ProfileState = 'loading' | 'success' | 'error';
+
+export const Profile_EditProfileScreen = ({
+  navigation,
+}: EditProfileScreenProps) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const {
+    authenticationType,
+    customerNumber,
+    phoneNumber: authPhoneNumber,
+  } = useAuthState();
+
+  const [customerProfile, setCustomerProfile] = useState<CustomerProfile>();
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [surname, setSurname] = useState('');
+  const [submissionStatus, setSubmissionStatus] = useState<
+    SubmissionStatus | undefined
+  >(undefined);
+  const [profileState, setProfileState] = useState<ProfileState>();
+
+  const isLoadingOrSubmittingProfile =
+    submissionStatus === 'loading' || profileState === 'loading';
+
+  const phoneNumber = parsePhoneNumber(
+    authPhoneNumber ?? '',
+  )?.formatInternational();
+
+  const onSubmit = async () => {
+    setSubmissionStatus('loading');
+
+    if (isValidEmail(email)) {
+      const {available} = await emailAvailable(email);
+      if (available || customerProfile?.email === email) {
+        try {
+          await updateProfile({
+            firstName,
+            surname,
+            email,
+          });
+          await fetchProfile();
+          setSubmissionStatus('submitted');
+        } catch {
+          setSubmissionStatus('submission-error');
+        }
+      } else {
+        setSubmissionStatus('unavailable-email');
+      }
+    } else {
+      setSubmissionStatus('invalid-email');
+    }
+  };
```

Comment:
We should use react-query now that we have added it to the app. And I think this will also be a little cleaner when using it, as loading states and such are handled by the lib.

Reply:
I nominate react query then for a team talk!
Will look at changing it here then 🌜

Reply:
Updated the code to use TanStack Query now 🚀